frontend/src/metabase/dashboard/containers/DashboardApp.unit.spec.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { Route } from "react-router";
+import { screen } from "@testing-library/react";
+
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+import { renderWithProviders, waitForElementToBeRemoved } from "__support__/ui";
+import DashboardApp from "metabase/dashboard/containers/DashboardApp";
+import {
+  createMockCollection,
+  createMockDashboard,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import {
+  setupCollectionItemsEndpoint,
+  setupCollectionsEndpoints,
+  setupDashboardEndpoints,
+} from "__support__/server-mocks";
+import { createMockDashboardState } from "metabase-types/store/mocks";
+import { BEFORE_UNLOAD_UNSAVED_MESSAGE } from "metabase/hooks/use-before-unload";
+
+const TEST_DASHBOARD = createMockDashboard({
+  id: 1,
+  name: "Example",
+});
+
+const TEST_COLLECTION = createMockCollection({
+  id: "root",
+});
+
+async function setup(user = createMockUser()) {
+  setupDashboardEndpoints(createMockDashboard(TEST_DASHBOARD));
+  setupCollectionsEndpoints([]);
+  setupCollectionItemsEndpoint(TEST_COLLECTION);
+
+  fetchMock.get("path:/api/bookmark", []);
+
+  const mockEventListener = jest.spyOn(window, "addEventListener");
+  const mockEvent = {
+    preventDefault: jest.fn(),
+  };
+  const callMockEvent = eventNames => {
```

Comment:
Maybe this is overkill, let me know

Reply:
nit: maybe we actually only need to setup `mockEventListener` here and then have a separate function for the call like:
```
const callMockEvent = (mockEventListener, eventName) => {
  const mockEvent = { preventDefault: jest.fn() };

  mockEventListener.mock.calls
      .filter(([event]) => eventNames.includes(event))
      .forEach(([_, callback]) => callback(mockEvent));

  return mockEvent;
};
```